src/App.tsx

```diff
@@ -15,61 +16,136 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-enum SortType {
+enum sortType {
   NONE,
-  ALPABET,
+  NAME,
   LENGTH,
 }
 
-// Use this function in the render method
-function getReorderedGoods(
-  goods: string[],
-  sortType: SortType,
-  isReversed: boolean,
-) {
-  // Not to mutate the original array
-  const visibleGoods = [...goods];
-
-  // Sort and reverse goods if needed
-  // ...
-
-  return visibleGoods;
-}
-
-// DON'T save goods to the state
 type State = {
   isStarted: boolean,
   isReversed: boolean,
-  sortType: SortType,
+  sortBy: sortType,
 };
 
-export const App = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    isStarted: false,
+    isReversed: false,
+    sortBy: sortType.NONE,
+  };
+
+  getStart = () => {
+    this.setState(state => ({
+      isStarted: !state.isStarted,
+    }));
+  };
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  sortByName = () => {
+    this.setState({ sortBy: sortType.NAME });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortBy: sortType.LENGTH });
+  };
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: sortType.NONE,
+    });
+  };
+
+  render(): React.ReactNode {
+    const { isReversed, isStarted, sortBy } = this.state;
+    const visibleProduct = [...goodsFromServer];
+
+    switch (sortBy) {
+      case sortType.NAME:
+        visibleProduct.sort(
+          (prod1, prod2) => prod1.localeCompare(prod2),
+        );
+        break;
+
+      case sortType.LENGTH:
+        visibleProduct.sort(
+          (prod1, prod2) => prod1.length - prod2.length,
+        );
+        break;
+
+      default:
+    }
+
+    if (isReversed) {
+      visibleProduct.reverse();
+    }
```

Comment:
move this logic to an external function, `render` shouldn't have a lot of logic in it